content/blog/en/introduction-to-testing-with-dojo.md

```diff
@@ -0,0 +1,151 @@
+
+## Assertion Templates
+
+The `Profile` widget we tested above has a property username that we can test against in the output. We could rewrite the entirety of the expected render output for each test (that's a lot of typing) or we could create an assertion to test against that would allow us to change the expected properties on each run. An important aspect of assertion templates is they ensure assertions are made for entire unit, which in our case is always the entire render output. It can often temping be to trigger an action and then only partially assert the section of the render output that the action is expected to affect. Partially asserting a widget's render output can be problematic as it means a test can "pass" but in fact the render output outside of the section asserted was affected and not caught.
+
+and lead to uncaught bugs  a widgets structure means that we don't know if the action of triggering the event handler had any unexpected affects on other areas of a widget's render output.
+
+To look at how assertion templates are used, we need to update the `Profile` widget's property API to make the `username` optional.
+
+> src/widgets/Profile.tsx
+
+```tsx
+import { tsx, create } from '@dojo/framework/core/vdom';
+
+import * as css from './styles/Profile.m.css';
+
+export interface ProfileProperties {
+	username?: string;
+}
+
+const factory = create().properties<ProfileProperties>();
+
+export default factory(function Profile({ properties }) {
+	const { username = 'Stranger' } = properties();
+	return <h1 classes={[css.root]}>{`Welcome ${username}!`}</h1>;
+});
+```
+
+Firstly the `assertionTemplate` needs to be imported into our test module. Next we need create a "base" assertion that represents the widget's default render output based on the properties API, for the `Profile` it is the `h1` node with the default `username`.
+
+```tsx
+const { describe, it } = intern.getInterface('bdd');
+import harness from '@dojo/framework/testing/harness';
+import { tsx } from '@dojo/framework/core/vdom';
+import assertionTemplate from '@dojo/framework/testing/assertionTemplate';
+
+import Profile from '../../../src/widgets/Profile';
+import * as css from '../../../src/widgets/styles/Profile.m.css';
+
+const baseAssertion = assertionTemplate(() =>
+  <h1 classes={[css.root]} "assertion-key"="welcome">Welcome Stranger!</h1>
+);
+
+describe('Profile', () => {
+	it('default renders correctly', () => {
+		const h = harness(() => <Profile />);
+		h.expect(baseAssertion);
+	});
+});
+```
+
+We can test against our base assertion like we did before. In our assertion template, we add an `assertion-key` property to the node so that we can update the template for additional test scenarios.
```

Comment:
```suggestion
We can continue testing against our base assertion. In our assertion template, we add an `assertion-key` property to the node so that we can update the template for additional test scenarios.
```